### PR TITLE
LPS-174454 Add POSHI automation | Master

### PR DESCRIPTION
**Pull request by alessandromalves**
This is only for master
https://issues.liferay.com/browse/LPS-174454
cc.: @david-gutierrez-mesa @Tim-Cao

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by alessandromalves** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/utilitypage/UtilityPages.testcase [hunk 1]
```diff
@@ -612,4 +612,71 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-174454. A new user without assigning permission cannot mark a Utility Page as default."
+	@priority = 4
+	test UserWithoutAssigningPermissionCannotMarkUtilityPageAsdefault {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/utilitypage/UtilityPages.testcase:617]: CannotMarkUtilityPageAsDefaultAsUserWithoutAssignDefaultPermission

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/utilitypage/UtilityPages.testcase [hunk 2]
```diff
@@ -612,4 +612,71 @@ definition {
...
+		property custom.properties = "feature.flag.LPS-170183=true";
+
+		task ("Given a page designer has an utility page") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/utilitypage/UtilityPages.testcase:620]: Given a user doesn't have assign default utility page permission

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/utilitypage/UtilityPages.testcase [hunk 3]
```diff
@@ -612,4 +612,71 @@ definition {
...
+
+			PagesAdmin.addUtilityPage(pageName = "Test Utility Page Name");
+
+			PageEditor.publish();
+		}
+
+		task ("When the Admin create an user without assigning permission") {
+			JSONRole.addRegularRole(
+				roleKey = "Utility Pages Permissions",
+				roleTitle = "Utility Pages Permissions");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/utilitypage/UtilityPages.testcase:633]: Regular Role Name

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/utilitypage/UtilityPages.testcase [hunk 4]
```diff
@@ -612,4 +612,71 @@ definition {
...
+			Navigator.gotoNavTab(navTab = "Utility Pages");
+
+			PagesAdmin.addUtilityPage(pageName = "Test Utility Page Name");
+
+			PageEditor.publish();
+		}
+
+		task ("When the Admin create an user without assigning permission") {
+			JSONRole.addRegularRole(
+				roleKey = "Utility Pages Permissions",
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/utilitypage/UtilityPages.testcase:632]: Regular Role Name

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/utilitypage/UtilityPages.testcase [hunk 5]
```diff
@@ -612,4 +612,71 @@ definition {
...
+				resourceAction = "ASSIGN_DEFAULT_LAYOUT_UTILITY_PAGE_ENTRY",
+				resourceName = "com.liferay.portal.kernel.model.Group",
+				roleTitle = "Utility Pages Permissions",
+				roleType = "regular");
+
+			JSONRole.assignRoleToUser(
+				roleTitle = "Utility Pages Permissions",
+				userEmailAddress = "userea@liferay.com");
+		}
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/utilitypage/UtilityPages.testcase:666]: task ("When the user navigates to the Pages admin")

**Review by Tim-Cao — COMMENTED** [review]
Hi @alessandromalves, would you mind adapting according to comments? Btw, please move the case to UtilityPagesWithPermissions.testcase file. Thanks

**Comment by alessandromalves** [comment]
ci:reopen

**Comment by alessandromalves** [comment]
It should work now. Thanks @Tim-Cao!

**Comment by alessandromalves** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/utilitypage/UtilityPagesWithPermissions.testcase [hunk 6]
```diff
@@ -0,0 +1,102 @@
+@component-name = "portal-wcm"
+definition {
+
+	property custom.properties = "feature.flag.LPS-170183=true";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/utilitypage/UtilityPagesWithPermissions.testcase:4]: Remove this one. See https://liferay.slack.com/archives/G03LYN0J8/p1677228508612729

**Review by Tim-Cao — COMMENTED** [review]
Hi @alessandromalves, feel free to run forward this one once you modify. Thanks

**Comment by alessandromalves** [comment]
ci:testLsf

**Comment by alessandromalves** [comment]
ci:test:sf

**Comment by alessandromalves** [comment]
ci:stop

**Comment by alessandromalves** [comment]
ci:stop

**Comment by alessandromalves** [comment]
ci:stop

**Comment by alessandromalves** [comment]
ci:forward

**Pull request closed without merging**